x/concentrated-liquidity/incentives.go

```diff
@@ -930,6 +930,18 @@ func (k Keeper) claimAllIncentivesForPosition(ctx sdk.Context, positionId uint64
 
 	supportedUptimes := types.SupportedUptimes
 
+	// Determine the qualifying balancer pool shares for the pools uptime accumulators
+	balancerPoolId, qualifyingBalancerShares, err := k.prepareBalancerPoolAsFullRange(ctx, position.PoolId, uptimeAccumulators)
+	if err != nil {
+		return sdk.Coins{}, sdk.DecCoins{}, err
+	}
```

Comment:
So we essentially repeat `prepareBalancerPoolAsFullRange` and `claimAndResetFullRangeBalancerPool` in the same message.

We already do this in `updatePoolUptimeAccumulatorsToNow`. What are the consequences of doing it twice?

Reply:
We need to do this the first time to update the accums to now and claim any outstanding rewards. The second time we call is needed to include the balancer pool liquidity in the forfeiture logic and then have the balancer pool immediately claim their share of the forfeited funds. If we don't do this, there would be no way to distribute forfeited funds to the balancer positions

Reply:
I will admit, reinvesting forfeited funds back into the accumulator has resulted in, at a minimum, a lot of mental overhead in reasoning what the expected values must be